src/shared_memory_manager.cc

```diff
@@ -346,12 +346,26 @@ SharedMemoryManager::GetMemoryInfo(
         std::string("Unable to find shared memory region: '" + name + "'")
             .c_str());
   }
+
+  // populate 'shm_mapped_addr'
+  size_t max_offset = 0;
```

Comment:
Is the logic any easier or more comprehensive if we consider `min_region_addr <= target_addr <= max_region_addr` bounds rather than what looks like the current `target_addr <= max_region_addr`

Reply:
I suppose `size_t` is unsigned, so any offset cannot be `< 0`. Then the `min_region_addr = target_addr`? Given `original_target_addr + (min_offset = 0) = original_target_addr = target_addr`, I think there is nothing to check for `min_region_addr` unless the `original_target_addr` is invalid.

Reply:
I think the above is not a clear explanation, let me put it differently:
Since all the variables here are unsigned, the sum of those variables can only be larger except when it overflows, which is why checking for `min_region_addr` is necessary. Assuming the `offset_` and `byte_size_` supplied during shm registration are valid (which should be checked during registration), then the maximum theoretical offset obtained from summing those three values cannot overflow, i.e. `max_offset = offset_ + byte_size_`. Thus, we have two good values to compare `offset > max_offset` and that will be sufficient for the check. I think this is a better check because it does not only ensure the `target_addr` is within range, but also ensures the computation of the `target_addr` did not overflow, i.e. `target_addr = mapped_addr_ + offset_ + offset`.

> what looks like the current `target_addr <= max_region_addr`

The solution is not comparing address, but validating offsets that sum to the address.